pkg/server/pubsub.go

```diff
@@ -33,16 +41,32 @@ func NewPubSubMessenger(topic *pubsub.Topic) *PubSubMessenger {
 
 // Send sends a pmap event to a Google Cloud PubSub topic.
 func (p *PubSubMessenger) Send(ctx context.Context, data []byte, attr map[string]string) error {
-	// TODO(#110): Need to add a LimitReader to read from processErr
-	// before adding it to attr and published by pubsub.
-	// https://cloud.google.com/pubsub/quotas#resource_limits.
-	result := p.topic.Publish(ctx, &pubsub.Message{
-		Data:       data,
-		Attributes: attr,
-	})
+	m := createPubsubMessageWithSizeLimit(data, attr)
 
+	result := p.topic.Publish(ctx, m)
 	if _, err := result.Get(ctx); err != nil {
-		return fmt.Errorf("pubsub: failed to get result returned from publish : %w", err)
+		return fmt.Errorf("pubsub failed to get result returned from publish : %w", err)
 	}
 	return nil
 }
+
+func createPubsubMessageWithSizeLimit(data []byte, attr map[string]string) *pubsub.Message {
+	d := data[:min(MaxTopicDataBytes, len(data))]
+
+	for key, value := range attr {
+		v := []byte(value)[:min(MaxTopicAttrValueBytes, len(value))]
```

Comment:
For one, I don't think we need check key size. For two, an oversized key should return error. Truncating key might cause key conflict.

Reply:
I think we are not truncating key, we only truncating values here? I kept the key as the same as it is.

Are you talking about not truncating value?

Reply:
My bad I misread it. Then why we still have `MaxTopicAttrKeyBytes`?

Reply:
Will remove it.